src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1541,6 +1541,27 @@ private static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSta
                 else
                 {
                     var property = sourceMethod.AssociatedSymbol as SourcePropertySymbol;
+                    if (!sourceMethod.IsExtern &&
+                        !sourceMethod.IsAbstract &&
+                        !sourceMethod.IsPartial &&
+                        !sourceMethod.IsExpressionBodied &&
+                        !sourceMethod.IsImplicitlyDeclared)
+                    {
+                        // avoid cascading on top of error ERR_AddRemoveMustHaveBody
+                        if (!(sourceMethod.AssociatedSymbol is SourceEventSymbol))
+                        {
+                            // avoid cascading on top of ERR_BadAsyncLacksBody
+                            if (!sourceMethod.IsAsync)
+                            {
+                                // avoid cascading on top of error ERR_AutoPropertyMustHaveXetAccessor
+                                if ((object)property == null || (!property.IsAutoProperty && (object)property.GetMethod != null && (object)property.GetMethod != null))
```

Comment:
The reason I split them into nested ifs is because I found them harder to read when merged. I'll probably leave as-is.